tools/events.php

```diff
@@ -0,0 +1,227 @@
+
+        foreach (self::AVAILABLE_OPTIONS as $commandName => $description) {
+            $spacingWidth = $width - Helper::width($commandName);
+            $this->line(
+                sprintf(
+                    '  <info>%s</info>%s%s',
+                    $commandName,
+                    str_repeat(' ', $spacingWidth),
+                    $description
+                )
+            );
+        }
+    }
+
+    /**
+     * Retrieve the columnWidth based on the commands text size
+     */
+    protected function getColumnWidth(array $commands): int
+    {
+        $widths = [];
+
+        foreach ($commands as $command) {
+            $widths[] = Helper::width($command);
+        }
+
+        return $widths ? max($widths) + 2 : 0;
+    }
+
+    /**
+     * List all events registered
+     */
+    protected function list(): void
+    {
+        $eventServiceProvider = app()
+            ->makeWith(
+                PKPEventServiceProvider::class,
+                ['app' => PKPContainer::getInstance()]
+            );
+
+        $events = [];
+
+        $rawEvents = $eventServiceProvider->getEvents();
+
+        foreach($rawEvents as $event => $listeners) {
+            $events[] = [$event, implode(', ', $listeners)];
+        }
+
+        $this->table(['Event', 'Listeners'], $events);
+    }
+
+    /**
+     * Clean the Event cached file
+     */
+    protected function clear(): void
+    {
+        $cacheManager = CacheManager::getManager();
+        $cacheManager->flush('event');
+
+        $this->getOutput()->success('Cache cleared!');
+    }
+
+    /**
+     * Create an Event cached file
+     */
+    protected function cache(): void
+    {
+        // Cleaning old file
+        $cacheManager = CacheManager::getManager();
+        $cacheManager->flush('event');
+
+        $eventServiceProvider = app()
+            ->makeWith(
+                PKPEventServiceProvider::class,
+                ['app' => PKPContainer::getInstance()]
+            );
+
+        // Forcing the cache rebuild
+        $eventServiceProvider->getEvents();
+
+        $filePath = $cacheManager->getFileCachePath();
+        $files = glob($filePath . DIRECTORY_SEPARATOR . 'fc-event*.php');
+        foreach ($files as $file) {
+            if (!file_exists($file)) {
+                $this->getOutput()->error('Cached file was not created');
+
+                return;
+            }
+        }
+
+        $this->getOutput()->success('Cache rebuilt!');
+    }
+
+    /**
+     * Parse and execute list event command
+     */
+    public function execute()
+    {
+        if (!isset(self::AVAILABLE_OPTIONS[$this->option])) {
+            throw new CommandNotFoundException(
+                sprintf('Option "%s" does not exist.', $this->option),
+                array_keys(self::AVAILABLE_OPTIONS)
+            );
+        }
+
+        $this->{$this->option}();
+    }
+}
+
+try {
+    $tool = new commandEvents($argv ?? []);
```

Comment:
@asmecher After seeing the #7171, I thought the command belongs to 4622, because he depends more on the EventServiceProvider caching.